### DISCUSSION TITLE
How does regex allowedVersions actually filter results?

### DISCUSSION THREAD

**Discussion by catostrophe** [renovatebot/renovate#20822] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.147.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

15.3.5-ee

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I have an image versioned as: `imagename:jdk-17.0.6-something`

I'm trying to apply a custom versioning scheme:
```
 "packageRules": [
    {
      "matchDatasources": "docker",
      "matchPackagePatterns": "my-jdk-image-mask",
      "matchCurrentValue": "/^jdk-/",
      "versioning": "regex:^jdk-(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)-(?<compatibility>.+)$",
      "allowedVersions": "/^jdk-17\\.0\\.[0-5]\\D/"
    }
  ]
```
If the current version is 17.0.3 and I want to allow versions strictly less than 17.0.6, then:
`"allowedVersions": "/^jdk-17\\.0\\.[0-5]\\D/"` doesn't work.
If I do `"allowedVersions": "/^jdk-17\\D/"`, it works but doesn't restrict versions and updates it to 17.0.6.

Trace logs don't contain info regarding how renovate applies allowedVersions regex to all possible versions.
 

### What you tried so far.

`"allowedVersions": "/^jdk-17\\.0\\.[0-5]\\D/"` doesn't work.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
The source code is here: https://github.com/renovatebot/renovate/blob/a72998724ef99bf2e887eedad7dd94c43bf61c83/lib/workers/repository/process/lookup/filter.ts#L64-L69

I don't see anything wrong with your regex, so can you reproduce the problem in a public repo using some public image with similar tag structure?
